tfx/benchmarks/beam_pipeline_benchmark_base.py

```diff
@@ -0,0 +1,198 @@
+
+    self.tfma_test_names = []
+    self.tfma_v2_test_names = []
+    self.tft_test_names = []
+
+    benchmark_base.set_cloud_dataflow_temp_loc(cloud_dataflow_temp_loc)
+
+  def _run_tfma_benchmarks(self):
+    tfma_benchmark_chicago_taxi = TFMABenchmarkChicagoTaxi(dataset=self.dataset)
+    self.wall_times["tfma"]["TFMABenchmarkChicagoTaxi.benchmarkMiniPipeline"] = tfma_benchmark_chicago_taxi.benchmarkMiniPipeline()
+
+  def _run_tfma_v2_benchmarks(self):
+    tfma_v2_benchmark_chicago_taxi = TFMAV2BenchmarkChicagoTaxi(dataset=self.dataset)
+    self.wall_times["tfma_v2"]["TFMAV2BenchmarkChicagoTaxi.benchmarkMiniPipelineUnbatched"] = tfma_v2_benchmark_chicago_taxi.benchmarkMiniPipelineUnbatched()
+    self.wall_times["tfma_v2"]["TFMAV2BenchmarkChicagoTaxi.benchmarkMiniPipelineBatched"] = tfma_v2_benchmark_chicago_taxi.benchmarkMiniPipelineBatched()
+
+  def _run_tft_benchmarks(self):
+    tft_benchmark_chicago_taxi = TFTBenchmarkChicagoTaxi(dataset=self.dataset)
+    self.wall_times["tft"]["TFTBenchmarkChicagoTaxi.benchmarkAnalyzeAndTransformDataset"] = tft_benchmark_chicago_taxi.benchmarkAnalyzeAndTransformDataset()
+
+  def _run_all_benchmarks(self):
+    self._run_tfma_benchmarks()
+    self._run_tfma_v2_benchmarks()
+    self._run_tft_benchmarks()
+
+  def _post_process(self):
+
+    # Add test names if dataset is empty
+    if self.wall_times_list == []:
+      test_names = ["Number of Replicas"]
+      for tf_module in self.wall_times:
+        for test in self.wall_times[tf_module]:
+          test_names.append(test)
+
+      self.wall_times_list.append(test_names)
+
+    # Add wall times to dataset
+    row = [self.num_workers]
+    for tf_module in self.wall_times:
+      for test, wall_time in self.wall_times[tf_module].items():
+        row.append(wall_time)
+
+    self.wall_times_list.append(row)
+    print(self.wall_times_list)
+
+  def _update_yaml(self, num_replicas):
+    with open(self.yaml_path) as f:
+      yaml_file = yaml.load(f)
+
+      yaml_file["spec"]["taskManager"]["replicas"] = num_replicas
+
+      with open(self.yaml_path, "w") as f:
+        yaml.dump(yaml_file, f)
+
+  def _write_to_csv(self, csv_filename):
+    with open(csv_filename, "w+") as  my_csv:
+      csv_writer = csv.writer(my_csv, delimiter=',')
+      csv_writer.writerows(self.wall_times_list)
+
+  def benchmarkFlinkOnK8s(self):
+    """Utilizes the flink-on-k8s-operator to run Beam pipelines"""
+
+    benchmark_base.set_beam_pipeline_mode("flink_on_k8s")
+    self.num_workers = self.min_num_replicas
+
+    while self.num_workers <= self.max_num_replicas:
+
+      # Update the .yaml file replicas value
+      self._update_yaml(self.num_workers)
+      time.sleep(10)
+
+      # Delete and reapply the kubectl clusters
+      subprocess.call("kubectl delete -f beam_flink_cluster.yaml", shell=True)
+      subprocess.call("kubectl delete -f beam_job_server.yaml", shell=True)
+      time.sleep(180)
```

Comment:
lolol I was conservative sleep time and I found it reliable up to 32 workers (haven't tested on 64). It'd be difficult to check in a loop since we'd have to parse kubectl output pretty